src/ucs/datastruct/callbackq.h

```diff
@@ -186,6 +210,43 @@ ucs_status_t ucs_callbackq_remove_safe(ucs_callbackq_t *cbq, ucs_callback_t cb,
 
 
 /**
+ * Add a callback to the slow path list.
+ * This can be used from any context and any thread, including but not limited to:
+ * - A callback can add another callback.
+ * - A thread can add a callback while another thread is dispatching callbacks.
+ *
+ * Important note: It is not allowed to add the same element more than once.
```

Comment:
I've removed the original check per @yosefe comment, because it made the functions to have linear dependency on the current number of slow path list elements. 
Another option would be introducing some debug flag in the slow element structure. Then this flag could be used for marking the element as already inserted in the slow path list. In this case the check would be O(1).  To implement this approach we should initialize the flag (as not present in the list) for each element, but the current API does not assume initializing the elements.